### PR TITLE
toc: reorganize TiCDC docs in left navigation

### PR DESCRIPTION
**Pull request by lilin90**
### What is changed, added or deleted? (Required)

- Moves TiCDC docs from **TiDB Tools** > [**TiCDC**](https://docs.pingcap.com/tidb/dev/ticdc-overview) to the first level in left navigation, to make it easier to find
- Renames TiCDC to **Stream Data** in the left navigation
- Combines the exiting first-level [**Integrate**](https://docs.pingcap.com/tidb/dev/integration-overview) with the TiCDC docs

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.5 (TiDB 8.5 versions)
- [ ] v8.4 (TiDB 8.4 versions)
- [ ] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: TOC.md [hunk 1]
```diff
@@ -169,11 +169,52 @@
     - [Migrate to a Downstream Table with More Columns](/migrate-with-more-columns-downstream.md)
     - [Filter Binlog Events](/filter-binlog-event.md)
     - [Filter DML Events Using SQL Expressions](/filter-dml-event.md)
-- Integrate
-  - [Overview](/integration-overview.md)
-  - Integration Scenarios
+- Stream Data
+  - [TiCDC Overview](/ticdc/ticdc-overview.md)
+  - [Deploy and Maintain](/ticdc/deploy-ticdc.md)
+  - Changefeeds
```
Comment (Oreoxmt) [inline TOC.md:175]:
```suggestion
  - Changefeed
```

**Review by Oreoxmt — APPROVED** [review]

File: TOC.md [hunk 2]
```diff
@@ -169,11 +169,52 @@
...
+    - [DDL Replication](/ticdc/ticdc-ddl.md)
+    - [Bidirectional Replication](/ticdc/ticdc-bidirectional-replication.md)
+    - [Data Integrity Validation for Single-Row Data](/ticdc/ticdc-integrity-check.md)
+    - [Data Consistency Validation for TiDB Upstream/Downstream Clusters](/ticdc/ticdc-upstream-downstream-check.md)
+    - [TiCDC Behavior in Splitting UPDATE Events](/ticdc/ticdc-split-update-behavior.md)
+  - Monitor and Alert
+    - [Monitoring Metrics Summary](/ticdc/ticdc-summary-monitor.md)
+    - [Monitoring Metrics Details](/ticdc/monitor-ticdc.md)
+    - [Alert Rules](/ticdc/ticdc-alert-rules.md)
+  - Integrate
```
Comment (benmeadowcroft) [inline TOC.md:194]: I think "Integration Scenarios" would explain this section better

File: TOC.md [hunk 3]
```diff
@@ -169,11 +169,52 @@
...
+  - [Deploy and Maintain](/ticdc/deploy-ticdc.md)
+  - Changefeed
+    - [Overview](/ticdc/ticdc-changefeed-overview.md)
+    - Create Changefeeds
+      - [Replicate Data to MySQL-compatible Databases](/ticdc/ticdc-sink-to-mysql.md)
+      - [Replicate Data to Kafka](/ticdc/ticdc-sink-to-kafka.md)
+      - [Replicate Data to Pulsar](/ticdc/ticdc-sink-to-pulsar.md)
+      - [Replicate Data to Storage Services](/ticdc/ticdc-sink-to-cloud-storage.md)
+    - [Manage Changefeeds](/ticdc/ticdc-manage-changefeed.md)
+    - [TiCDC Client Authentication](/ticdc/ticdc-client-authentication.md)
```
Comment (benmeadowcroft) [inline TOC.md:183]: This is not changefeed specific so should be in the Reference section

File: TOC.md [hunk 4]
```diff
@@ -169,11 +169,52 @@
...
+    - [Log Filter](/ticdc/ticdc-filter.md)
+    - [DDL Replication](/ticdc/ticdc-ddl.md)
+    - [Bidirectional Replication](/ticdc/ticdc-bidirectional-replication.md)
+    - [Data Integrity Validation for Single-Row Data](/ticdc/ticdc-integrity-check.md)
```
Comment (benmeadowcroft) [inline TOC.md:187]: I think this is a good candidate for the Reference section

File: TOC.md [hunk 5]
```diff
@@ -169,11 +169,52 @@
...
+    - [Data Consistency Validation for TiDB Upstream/Downstream Clusters](/ticdc/ticdc-upstream-downstream-check.md)
```
Comment (benmeadowcroft) [inline TOC.md:188]: I think this is a good candidate for the Reference section

File: TOC.md [hunk 6]
```diff
@@ -169,11 +169,52 @@
...
+    - [TiCDC Behavior in Splitting UPDATE Events](/ticdc/ticdc-split-update-behavior.md)
```
Comment (benmeadowcroft) [inline TOC.md:189]: I think this is a good candidate for the Reference section

**Review by benmeadowcroft — CHANGES_REQUESTED** [review]

**Comment by lilin90** [comment]
@benmeadowcroft Thanks for your suggestions! All the suggestions have been accepted via https://github.com/pingcap/docs/pull/19365/commits/648fd005b35c3ce1f0b2d659c9649c46d2d96ef6.

**Review by benmeadowcroft — APPROVED** [review]

**Pull request merged**